### PR TITLE
Tidy up validate-modules ignores for cloud/google modules

### PR DESCRIPTION
**Pull request by russoz**
##### SUMMARY
Tidy up on sanity tests' ignore files, removing lines with validation-modules for these modules.

The focus was to pass the tests so no code was changed other than adjusting argument_spec for missing types or inconsistencies with the docs. The priority has been not to change any functionality, but to ensure that code and docs are synchronized.

Some lines were added back to the files, but a total of 235 lines were removed (among 2.9, 2.10 and 2.11).

##### ISSUE TYPE
- Docs Pull Request

##### COMPONENT NAME
plugins/modules/cloud/google/gce.py
plugins/modules/cloud/google/gc_storage.py
plugins/modules/cloud/google/gcspanner.py
plugins/modules/cloud/google/gcp_url_map.py
plugins/modules/cloud/google/gcpubsub.py
plugins/modules/cloud/google/gcpubsub_info.py
plugins/modules/cloud/google/gcp_target_proxy.py
plugins/modules/cloud/google/gcp_healthcheck.py
plugins/modules/cloud/google/gcp_forwarding_rule.py
plugins/modules/cloud/google/gcp_backend_service.py
plugins/modules/cloud/google/gce_tag.py
plugins/modules/cloud/google/gce_snapshot.py
plugins/modules/cloud/google/gce_pd.py
plugins/modules/cloud/google/gce_net.py
plugins/modules/cloud/google/gce_mig.py
plugins/modules/cloud/google/gce_lb.py
plugins/modules/cloud/google/gce_labels.py
plugins/modules/cloud/google/gce_instance_template.py
plugins/modules/cloud/google/gce_img.py
plugins/modules/cloud/google/gce_eip.py
plugins/modules/cloud/google/gcdns_zone.py
plugins/modules/cloud/google/gcdns_record.py

### CODE REVIEW

File: tests/sanity/ignore-2.9.txt [hunk 1]
```diff
@@ -36,91 +36,18 @@ plugins/modules/cloud/dimensiondata/dimensiondata_vlan.py validate-modules:doc-c
 plugins/modules/cloud/dimensiondata/dimensiondata_vlan.py validate-modules:doc-missing-type
 plugins/modules/cloud/dimensiondata/dimensiondata_vlan.py validate-modules:parameter-type-not-in-doc
 plugins/modules/cloud/docker/docker_container.py use-argspec-type-path # uses colon-separated paths, can't use type=path
-plugins/modules/cloud/google/gc_storage.py validate-modules:doc-choices-do-not-match-spec
-plugins/modules/cloud/google/gc_storage.py validate-modules:doc-default-does-not-match-spec
-plugins/modules/cloud/google/gc_storage.py validate-modules:doc-missing-type
-plugins/modules/cloud/google/gc_storage.py validate-modules:parameter-type-not-in-doc
-plugins/modules/cloud/google/gc_storage.py validate-modules:undocumented-parameter
-plugins/modules/cloud/google/gcdns_record.py validate-modules:deprecation-mismatch
```
Comment (felixfontein) [inline tests/sanity/ignore-2.9.txt:44]: You need to keep the deprecation-mismatch lines in ignore-2.9.txt.

Comment (russoz) [reply tests/sanity/ignore-2.9.txt:44]: how can I force ansible-test to run against 2.9 locally? my regular cmd line is:
```
$ ansible-test sanity --docker default -v plugins/modules/cloud/google/gc_storage.py
```
and from what I can tell out of the output, it runs against ansible 2.10 and I do not see any obvious parameter in the `--help` output about that.

**Comment by russoz** [comment]
the tally went down to only 208 lines overall...

File: tests/sanity/ignore-2.9.txt (hunk 1 above)
Comment (felixfontein) [reply tests/sanity/ignore-2.9.txt:44]: You need to install Ansible 2.9 for that (using a venv would be a good idea). ansible-test comes with ansible(-base), so there's no verison selector.

**Comment by russoz** [comment]
the `metadata` parameter is in a pitiful state in `gce_instance_template.py`, will need to rollback a couple of ignores for that as well... seems to be the last one, but...

**Comment by felixfontein** [comment]
It is indeed! I guess the best solution (not for this PR though) would be to re-add `type: dict`, also adjust the argspec accordingly, and get rid of the hacks here: https://github.com/ansible-collections/community.general/blob/41a998cf3024391223008ae0ae644b73b7de71a6/plugins/modules/cloud/google/gce_instance_template.py#L408-L416

File: plugins/modules/cloud/google/gce.py [hunk 2]
```diff
@@ -21,68 +21,83 @@
...
         U(https://cloud.google.com/sdk/gcloud/reference/compute/instances/create),
         --scopes section for detailed information)
-    choices: [
-      "bigquery", "cloud-platform", "compute-ro", "compute-rw",
-      "useraccounts-ro", "useraccounts-rw", "datastore", "logging-write",
-      "monitoring", "sql-admin", "storage-full", "storage-ro",
-      "storage-rw", "taskqueue", "userinfo-email"
-    ]
+      - >
+        Available choices are: [
```
Comment (felixfontein) [inline plugins/modules/cloud/google/gce.py:64]:
```suggestion
        Available choices are:
```
Same for the closing `]`.

File: plugins/modules/cloud/google/gce_instance_template.py [hunk 3]
```diff
@@ -75,19 +89,21 @@
...
         U(https://cloud.google.com/sdk/gcloud/reference/compute/instances/create),
         --scopes section for detailed information)
-    choices: [
-      "bigquery", "cloud-platform", "compute-ro", "compute-rw",
-      "useraccounts-ro", "useraccounts-rw", "datastore", "logging-write",
-      "monitoring", "sql-admin", "storage-full", "storage-ro",
-      "storage-rw", "taskqueue", "userinfo-email"
-    ]
+      - >
+        Available choices are: [
```
Comment (felixfontein) [inline plugins/modules/cloud/google/gce_instance_template.py:102]:
```suggestion
        Available choices are:
```
Same for the closing `]`.

File: plugins/modules/cloud/google/gcp_healthcheck.py [hunk 4]
```diff
@@ -41,72 +41,85 @@
...
         U(https://cloud.google.com/sdk/gcloud/reference/compute/instances/create),
         --scopes section for detailed information)
-    choices: [
-      "bigquery", "cloud-platform", "compute-ro", "compute-rw",
-      "useraccounts-ro", "useraccounts-rw", "datastore", "logging-write",
-      "monitoring", "sql-admin", "storage-full", "storage-ro",
-      "storage-rw", "taskqueue", "userinfo-email"
-    ]
+      - >
+        Available choices are: [
```
Comment (felixfontein) [inline plugins/modules/cloud/google/gcp_healthcheck.py:112]:
```suggestion
        Available choices are:
```
Same here.

**Review by felixfontein — COMMENTED** [review]
Rest looks good. Also looks like this modules are (partially) in a pretty sorrow state. I wonder if they actually still work...

**Comment by russoz** [comment]
@felixfontein agree on the metadata approach... not this PR.

**Review by felixfontein — APPROVED** [review]
LGTM

**Comment by felixfontein** [comment]
@russoz thanks for fixing this!

**Pull request merged**